### PR TITLE
Merge 2.4 to `main`

### PR DESCRIPTION
**Pull request by fwcd**
This merges 2.4 to `main`. Leaves all of the translation files untouched (I hope that's how this is done?)

### CODE REVIEW

**Comment by daschuer** [comment]
> This merges 2.4 to main. Leaves all of the translation files untouched (I hope that's how this is done?)

Correct. 

Did you resolve other conflicts?

**Comment by fwcd** [comment]
Yes

**Comment by fwcd** [comment]
CI passes, can this be merged?

File: CMakeLists.txt [hunk 1]
```diff
@@ -757,7 +757,6 @@ add_library(mixxx-lib STATIC EXCLUDE_FROM_ALL
   src/effects/backends/builtin/moogladder4filtereffect.cpp
   src/effects/backends/builtin/parametriceqeffect.cpp
   src/effects/backends/builtin/phasereffect.cpp
-  src/effects/backends/builtin/pitchshifteffect.cpp
```
Comment (ronso0) [inline CMakeLists.txt:760]: ??

Comment (ronso0) [reply CMakeLists.txt:760]: ah, it's added conditionally.

Comment (fwcd) [reply CMakeLists.txt:760]: Moved under `if(RUBBERBAND)` in #12661

File: res/Mixxx-Keyboard-Shortcuts.png
Comment (ronso0) [inline res/Mixxx-Keyboard-Shortcuts.png:1]: Ups, this slipped in by accident.
I'll remove it and merge to main after this has been merged.

File: CMakeLists.txt [hunk 2]
```diff
@@ -2665,7 +2670,7 @@ endif()
 option(DEBUG_ASSERTIONS_FATAL "Fail if debug become true assertions" OFF)
 if(DEBUG_ASSERTIONS_FATAL)
   target_compile_definitions(mixxx-lib PUBLIC MIXXX_DEBUG_ASSERTIONS_FATAL MIXXX_DEBUG_ASSERTIONS_ENABLED)
-  if(QML)
+  if(QT6)
```
Comment (fwcd) [inline CMakeLists.txt:2673]: Ah looks like this should probably be

```suggestion
  if(QML)
```

**Comment by fwcd** [comment]
Did another merge, to get #12687 included.

**Comment by ronso0** [comment]
I suggest to keep this as is, not extend it further so it remains somewhat reviewable and gets merged soonish.
Otoh, I haven't followed the cmake/iOS changes so I don't know what can go wrong.

I already used this to rebase my personal branch, but that doesn't mean much I guess since I only build on Linux.

**Comment by fwcd** [comment]
There are not really a lot of CMake changes (#12661, #12662, #12673 OTOH, but all of them have already been reviewed). Sadly, GitHub doesn't offer a good way to visualize just the conflict resolutions (i.e. what has changed apart from the "automatic" merge, which, in this case, is mostly just merging together a few adjacent methods in the star rating widgets, if I remember right)

**Comment by fwcd** [comment]
The more "drastic" CMake changes for iOS support are coming in #12672 and only target `main` anyway

**Comment by ronso0** [comment]
Well then, CI passes and if you're confident nothing broke, we can merge. Nevertheless I'd appreciate a second pair of eyes here.

**Comment by fwcd** [comment]
Did another quick local smoke-test, there are at least no obvious regressions. I agree that the large merge diffs are hard to review, though I can't really speak on most of the changes apart from what I've been involved in.

**Comment by daschuer** [comment]
I have adjusted the CMakeLists.txt at one point and updated with the latest 2.4 branch here: 
https://github.com/mixxxdj/mixxx/commit/14c227094a9df1761aeb9bea62d77d2cbebeb11c

Thank you for the work.

**Pull request closed without merging**